### ISSUE TITLE
[Mobile Payments] Weird payment/connection flow state

### ISSUE THREAD

**Issue by malinajirka** [woocommerce/woocommerce-ios#6612]
**Describe the bug**
When I attempt to collect a payment on a store with an empty store address and I switch to a different store IPP gets into a weird state.

**To Reproduce**
1. Switch to a site with an empty store address
2. Open detail of an unpaid order
3. Tap on collect payment
4. Notice "Missing store address" dialog is shown
5. Switch to a store with working IPP
6. Enter IPP settings
7. Connect to a reader
8. Disconnect from a reader 🔴 Notice the first tap on disconnect doesn't work -> try again
9. Try connect to a reader - 🔴  notice the connection keeps failing and `card_reader_connection_failed` - `Sorry, this payment couldn’t be processed` is tracked even though you are trying to connect to a reader not accept any payments.

There are more scenarios in which the app behaves unexpectedly - I think the key here is that when "missing store address" is shown, the app gets into a weird state - it doesn't cancel the payment flow perhaps? When I try switching stores and collecting a payment, the first attempt to collect the payment always fails.

**Mobile Environment**
Please include:
- Device: SIMULATOR
- iOS version: iPhone 13
- WooCommerce iOS version: dev `824da784749c47f4cac2659d6620b86e7ff73286`

**Comment by ctarda** [comment]
> it doesn't cancel the payment flow perhaps

This seems very likely.

**Comment by toupper** [comment]
Hi, @malinajirka Thanks for reporting the issue. I tried to reproduce it but without success. I attached a video of the flow, please let me know if there is something I am missing to reproduce the issue:

https://user-images.githubusercontent.com/1864060/162754276-cfdf147d-a64a-45de-8c8f-cfdd1171e2d0.mp4

**Comment by malinajirka** [comment]
@toupper Thanks for looking into it! At first I didn't see any difference between our testing steps. However, I've played with it a bit more and I think I found one more condition that needs to be met. One of the store needs to be using WCPay and the other Stripe Extension. I'm able to reproduce it in 100% cases. In case you still have troubles, perhaps, it's related to my stores? I could invite you, let me know.


https://user-images.githubusercontent.com/2261188/162873619-6f042941-1f84-45d1-9d17-eead7ab2903b.mp4
